cmd/gpu_plugin/rm/gpu_plugin_resource_manager.go

```diff
@@ -0,0 +1,417 @@
+// Copyright 2021 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rm
+
+import (
+	"context"
+	"os"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	dpapi "github.com/intel/intel-device-plugins-for-kubernetes/pkg/deviceplugin"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+	podresourcesv1 "k8s.io/kubelet/pkg/apis/podresources/v1"
+	"k8s.io/kubernetes/pkg/kubelet/apis/podresources"
+)
+
+const (
+	deviceType       = "i915"
+	fullResourceName = "gpu.intel.com/" + deviceType
+	monitorID        = "all"
+
+	gasTSAnnotation   = "gas-ts"
+	gasCardAnnotation = "gas-container-cards"
+
+	grpcAddress    = "unix:///var/lib/kubelet/pod-resources/kubelet.sock"
+	grpcBufferSize = 4 * 1024 * 1024
+	grpcTimeout    = 5 * time.Second
+
+	retryTimeout = 1 * time.Second
+)
+
+// Errors.
+var (
+	errZeroPending = zeroPendingErr{}
+	errRetry       = retryErr{}
+)
```

Comment:
These globals are not needed.

Reply:
The habit has come from linting. Granted, this error is not enabled for this project, but just so that you know where it comes from: "linter err113: do not define dynamic errors, use wrapped static errors instead"

But yeah, since we don't have that enabled I might as well do it like the rest of the plugins do it, which is using merrily dynamic errors.

Reply:
I guess this stems from the pre `error.Is()` era when errors didn't carry stack traces and simple comparison could be used to check their types.